mono/metadata/image.c

```diff
@@ -1087,16 +1091,25 @@ typedef struct {
 	const char guid [40];
 } ProblematicAssembly;
 
-const char *bad_assembly_names = {
-	"System.Runtime.InteropServices.RuntimeInformation",
+const char *bad_assembly_names[] = {
+	"System.Runtime.InteropServices.RuntimeInformation.dll",
+	"System.Globalization.Extensions.dll",
+	"System.IO.Compression.dll",
+	"System.Net.Http.dll",
+	"System.Text.Encoding.CodePages.dll"
 };
 
-static const ProblematicAssembly bad_assemblies [1] = {
-	{
-		.hash = 0xD07383BB,
-		.assembly_name = SYS_RT_INTEROP_RUNTIME_INFO,
-		"DD91439F-3167-478E-BD2C-BF9C036A1395"
-	}
+#define BAD_ASSEMBLY(HASH, NAME, GUID, VER_STR)	{ .hash = HASH, .assembly_name = NAME, .guid = GUID }
+
+static const ProblematicAssembly bad_assemblies [] = {
```

Comment:
We need more versions that just the latest one e.g. for System.Net I think we need at least 4.3.0, 4.1.1, 4.1.0, 4.0.0

note: 4.0.0.0 is the most popular